### PR TITLE
CD-148072 Add ability to listen to multiple consumers from a single f…

### PR DESCRIPTION
**Pull request by johnny-lai**
…etcher thread

- [x] @johnwu96822 
- [x] @edk
@supriya fyi

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -519,6 +519,37 @@ end
...
 
+You can also consume from multiple consumer groups with a single thread using `Consumers`.
+
+```ruby
+require "kafka"
+
+kafka = Kafka.new(["kafka1:9092", "kafka2:9092"])
+
+# Get a group of consumers
+consumers = kafka.consumers()
```
Comment (edk) [inline README.md]: may want to drop the `()` at the end

File: lib/kafka/client.rb [hunk 2]
```diff
@@ -395,6 +395,27 @@ def consumer(
       )
     end
 
+    def consumers(
+        fetcher_max_queue_size: 100
```
Comment (edk) [inline lib/kafka/client.rb]: indent seems off.

**Comment by johnny-lai** [comment]
fyi, https://github.com/coupa/kafque/pull/8 is where this will be used

File: lib/kafka/consumers.rb [hunk 3]
```diff
@@ -0,0 +1,277 @@
...
+
+    # Add the consumer to the group to fetch messages for.
+    #
+    # @param consumer [Consumer] the consumer to fetch messages for.
+    # @return [self]
+    def add(consumer)
+      group_id = consumer.group_id
+
+      # Remove existing if any
+      delete(group_id: group_id) if @consumers.key?(group_id)
```
Comment (johnwu96822) [inline lib/kafka/consumers.rb]: Why delete when all the values are assigned below?

**Comment by edk** [comment]
👍

**Comment by johnwu96822** [comment]
👍

**Comment by johnny-lai** [comment]
[lockit v4.9.0] merged via Lockit CLI on 2019-02-25 17:23:08.575112 -0800 PST m=+9.827013014

File: lib/kafka/consumers.rb (hunk 3 above)
Comment (johnny-lai) [reply lib/kafka/consumers.rb]: A consumer stores which topics it is subscribed to. If you add a consumer with an existing group, this will replace it.

**Pull request merged**